site/app/controllers/admin/ConfigurationController.php

```diff
@@ -4,8 +4,49 @@
 
 use app\controllers\AbstractController;
 use app\libraries\FileUtils;
+use app\controllers\forum\ForumController;
 
 class ConfigurationController extends AbstractController {
+
+    private function isValidCategories($inputCategoriesIds = -1, $inputCategoriesName = -1){
```

Comment:
This is actually in a different class in the refactor, but can we make this static and put it somewhere as a helper function rather than private here? We can pass $core as a parameter. This would then be able to be called from both here and ForumController.

Reply:
To follow up on our conversation from before, this function already does exist and has been moved as part of the refactor effort. It resides in `site/app/libraries/ForumUtils.php` you should be able to reference this from here instead of copying this function (`$rows` is the categories from the query result). Let me know if this doesn't work.